cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/worker/BasicWorkerContext.java

```diff
@@ -85,9 +87,10 @@ public BasicWorkerContext(WorkerSpecification spec, Program program, RunId runId
                             DatasetFramework datasetFramework,
                             TransactionSystemClient transactionSystemClient,
                             DiscoveryServiceClient discoveryServiceClient,
-                            StreamWriterFactory streamWriterFactory) {
+                            StreamWriterFactory streamWriterFactory,
+                            @Nullable String adapterName) {
```

Comment:
My branch has the whole `AdapterSpecification` passed to the worker context. Can you hold this change for a bit longer today? I am forming the PR.